migrations/20240823093832_init.sql

```diff
@@ -0,0 +1,51 @@
+-- Metadata about on-chain transactions
+-- Used to establish order of observed roots
+-- And to fetch latest block number from which to sync
+CREATE TABLE tx (
+    -- This id exists for the purpose of ordering these txs
+    -- and linking different tables together
+    -- the ids are strictly increasing, but they don't
+    -- maintain strict ordering across chains/contracts
+    id BIGSERIAL PRIMARY KEY,
+    chain_id BIGINT NOT NULL,
+    block_number BIGINT NOT NULL,
+    tx_hash BYTEA NOT NULL
+);
+
+-- The canonical chain updates
+CREATE TABLE updates (
+    id BIGSERIAL PRIMARY KEY,
+    tx_id BIGINT NOT NULL,
+
+    pre_root BYTEA NOT NULL UNIQUE,
+    post_root BYTEA NOT NULL UNIQUE,
+
+    FOREIGN KEY (tx_id) REFERENCES tx (id)
+);
+
+-- Table to monitor roots on each chain
+CREATE TABLE roots (
```

Comment:
This table essentially replaces the `chain_state` mapping we used to have in `WorldTree` struct.

In order to be HA compatible we store the whole history of observed roots, instead of only the latest one on each chain.